apps/dashboard/src/components/primitives/field-editor/variable-popover/variable-popover.tsx

```diff
@@ -0,0 +1,257 @@
+                  <Select
+                    value=""
+                    onValueChange={(value) => {
+                      if (value) {
+                        handleTransformerToggle(value);
+                        setSearchQuery('');
+                      }
+                    }}
+                  >
+                    <SelectTrigger className="!text-paragraph-xs text-text-soft !h-7 min-h-0 !p-1.5">
+                      <SelectValue placeholder="Add a modifier..." />
+                    </SelectTrigger>
+                    <SelectContent
+                      onCloseAutoFocus={(e) => {
+                        e.preventDefault();
+                        searchInputRef.current?.focus();
+                      }}
+                      className="max-h-[400px]"
+                      align="start"
+                    >
+                      <div className="p-2">
+                        <InputField size="fit" className="min-h-0">
+                          <Input
+                            ref={searchInputRef}
+                            value={searchQuery}
+                            onChange={(e) => {
+                              e.stopPropagation();
+                              setSearchQuery(e.target.value);
+                            }}
+                            className="h-7 text-sm"
+                            placeholder="Search modifiers..."
+                            autoFocus
+                            onKeyDown={(e) => {
+                              e.stopPropagation();
+                              if (e.key === 'Escape') {
+                                setSearchQuery('');
+                              }
+                            }}
+                          />
+                        </InputField>
+                      </div>
+                      <div className="max-h-[350px] overflow-y-auto px-1">
+                        {getFilteredTransformers(searchQuery).length === 0 ? (
+                          <div className="text-text-soft flex flex-col items-center justify-center gap-2 p-4 text-center">
+                            <span className="text-sm">
+                              {searchQuery ? 'No modifiers found' : 'All modifiers have been added'}
+                            </span>
+                            {searchQuery && <span className="text-xs">Try searching for different terms</span>}
+                          </div>
+                        ) : (
+                          getFilteredTransformers(searchQuery).map((transformer) => (
+                            <SelectItem
+                              key={transformer.value}
+                              value={transformer.value}
+                              className="relative [&>*:first-child]:p-0"
+                            >
+                              <TransformerItem transformer={transformer} />
+                            </SelectItem>
```

Comment:
the scroll does not work

https://github.com/user-attachments/assets/ae49681f-b87e-4090-b713-c205286851e8